ms2deepscore/data_generators.py

```diff
@@ -228,7 +237,7 @@ def __data_generation(self, spectrum_pairs: Iterator[SpectrumPair]):
             for i_spectrum, spectrum in enumerate(pair):
                 idx, values = self._data_augmentation(spectrum.binned_peaks)
                 X[i_spectrum][i_pair, idx] = values
-            y[i_pair] = self.reference_scores_df[pair[0].get('inchikey')][pair[1].get('inchikey')]
+            y[i_pair] = self.reference_scores_df[pair[0].get("inchikey")[:14]][pair[1].get("inchikey")[:14]]
```

Comment:
Now at every inchikey-based retrieval from `reference_scores_df` or `binned_spectrums` (this line is an example), we transform inchikey to inchikey14. Can we not transform the inchikeys once for both the reference_scores_df and binned_spectrums when we initialize the class?

Reply:
And then cleanest would be to do it in a method `transform_data_to_inchikey14` or something.

Reply:
I will add that change to the `reference_score_df`, but not to the `binned_spectrums` because the later contains a shallow copy pointing to the original spectrums (and for other purposes the full inchikey might be needed). 
(I agree that the `[:14]` additions are a bit ugly, but I would keep it for now, I think)

Reply:
Indeed, it's good to keep the original Inchikeys as well!